javascripts/default.js

```diff
@@ -476,4 +481,196 @@ NodeList.prototype.on = NodeList.prototype.addEventListener = (function(name, fn
     data[name] = window[name];
     checker(name);
   });
+
+
+  //
+  // TESTING
+  //
+
+  function fetch_data(name_to_path, callback) {
+    var data = {};
+    Object.keys(name_to_path).forEach(function(name) { data[name] = undefined; });
+    Object.keys(name_to_path).forEach(function(name) {
+      fetch(name_to_path[name]).then(function(r){ return r.json(); }).then(function(json) {
+        data[name] = json;
+        for (var p in data) {
+          if (typeof(data[p]) === 'undefined') {
+            // not all finished, ignore
+            return;
+          }
+        }
+        callback(data);
```

Comment:
callback will be invoked N times where N is the number of keys in name_to_path... is this by design?

Reply:
good catch. fixed.